src/App.tsx

```diff
@@ -10,33 +11,34 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTabId, setSelectedTab] = useState('tab-1');
+
+  useEffect(() => {
+    if (!tabs.map(tab => tab.id).includes(selectedTabId)) {
+      setSelectedTab('tab-1');
+    }
+  }, [selectedTabId]);
+
+  const onTabSelected = (newTab: {
```

Comment:
The 'onTabSelected' function does not check if the newTab is different from the currently selected tab. If it is the same, there is no need to call 'setSelectedTab', saving unnecessary rerenders.

add if check here e.g:

```
if (newTab.id !== selectedTabId) {
      onTabSelected(newTab.id);
}
```